modules/local/custom/write_reference_data/main.nf

```diff
@@ -0,0 +1,22 @@
+process WRITE_REFERENCE_DATA {
```

Comment:
This is used specifically to publish all reference data required for an analysis. The idea is to download reference data required for execution, which is important where a user wants to run an analysis repeatedly for different samples or run oncoanalyser offline.

The feature involves running the existing prepare reference data component (index building, archive decompressing, etc) and then passing *all* reference data files to this process so that they are published. Since not all reference files are consistently passed through a process during the prepare reference stage (e.g. the reference genome FAI), there must be an additional process used to publish these files. This is the role (and only role) that WRITE_REFERENCE_DATA has, hence there is no actual code executed in the script block.

I decided to pass every reference file to WRITE_REFERENCE_DATA in order to publish them if requested by the user rather than attempting to individually + conditionally configure several processes to do the same. I find this way is simpler and reduces maintenance burden

Reply:
This is clearly nice to have. It would in addition be good to move the reference data to igenomes in the long run. Reason being that people running offline tend to set up the reference data and let others run the pipeline.

Reply:
Created an issue to continue this piece of work:

- https://github.com/nf-core/oncoanalyser/issues/80